### DISCUSSION TITLE
Unable to render OpenAPI.json from an APIPlatform project using Twig Template

### DISCUSSION THREAD

**Discussion by valentimarco** [scalar/scalar#2926] [Q&A]
I have a ApiPlatform project where I want to add scalar by replacing the swagger template with a custom one.
I modify [this twig template](https://github.com/api-platform/core/blob/bd6a57c4c0b38d6f880a5bd79031a87033f707e6/src/Symfony/Bundle/Resources/views/SwaggerUi/index.html.twig#L100) and was able to render scalar (The layout) so after,  i injected the openapi.json generated from ApiPlatoform but don't render anything besides the layout(No errors in consoles)
Here the custom snippet:
```html
<head>
    <title>Scalar API Reference</title>
    <meta charset="utf-8"/>
    <meta
            name="viewport"
            content="width=device-width, initial-scale=1"/>
    {% block head_javascript %}
        {# json_encode(65) is for JSON_UNESCAPED_SLASHES|JSON_HEX_TAG to avoid JS XSS #}
        <script id="swagger-data" type="application/json">{{ swagger_data|json_encode(65)|raw }}</script>
    {% endblock %}
</head>
<body>
<!-- Need a Custom Header? Check out this example https://codepen.io/scalarorg/pen/VwOXqam -->
<script
        id="api-reference"
        type="application/json">
</script>
<script src="https://cdn.jsdelivr.net/npm/@scalar/api-reference"></script>
<script>
    const configuration = {
        hideModels: true
    }
    const doc = document.getElementById("api-reference")
    doc.dataset.configuration = JSON.stringify(configuration)
    const openApiJson = JSON.parse(document.getElementById("swagger-data").innerText)
    doc.innerText = JSON.stringify(openApiJson.spec)
</script>
</body>
```

Ref to where the json is created:  [SwaggerUiProcessor](https://github.com/api-platform/core/blob/bd6a57c4c0b38d6f880a5bd79031a87033f707e6/src/Symfony/Bundle/SwaggerUi/SwaggerUiProcessor.php#L93)

P.S i have tried the same openapi json in the scalar editor and works just fine. Also tried to put manully but don't work :(

**Comment by marclave** [comment]
hey @valentimarco ! thanks for sharing the code, if you're able to share a repo that i can checkout i can investigate : )

**Reply by valentimarco** [reply]
Unfortunally Is a private repo :(

**Reply by marclave** [reply]
ah! okay, any chance you could make a public repository that has the bare minimum dev environment?

i wasn't able to recreate this issue.

also happy to do a video call to go over the issues https://cal.com/scalar :)

**Reply by valentimarco** [reply]
Ok i did it xd
When creating the demo i tried some stuff and manage to work although i needed to change the uri path and create a custom controller to handle the request.
Here is the demo i created:
https://github.com/valentimarco/demo-with-scalar

**Reply by valentimarco** [reply]
In the main project i get some error:
1. ~~related to some urls with uri variables~~ fixed by inserting title, description and version in `api_platform.yaml`
2. `getExampleFromSchema] Unknown property type "undefined".` 
3. ~~The initial description of the docs (where we have tittle, description etc..) is not render~~ same as 1.

**Reply by valentimarco** [reply]
Update: i was able to render scalar using the old idea. Here is how i did it:
1. Create the twig replacement under `templates/bundles/ApiPlatformBundle/SwaggerUi/index.html.twig`
2. Copy the html below:
```html
{# templates/bundles/ApiPlatformBundle/SwaggerUi/index.html.twig #}
{% set openApiJson = swagger_data["spec"] %}
<!doctype html>
<html>
<head>
    <title>Scalar API Reference</title>
    <meta charset="utf-8"/>
    <meta
            name="viewport"
            content="width=device-width, initial-scale=1"/>
</head>
<body>
<!-- Need a Custom Header? Check out this example https://codepen.io/scalarorg/pen/VwOXqam -->
<script
        id="api-reference"
        type="application/json">{{ openApiJson| json_encode |raw}}</script>
<script src="https://cdn.jsdelivr.net/npm/@scalar/api-reference"></script>
</body>
</html>
```
Remember to set under `api_platform.yaml` the property `tittle`, `description` and `version`
